src/System.Private.CoreLib/shared/System/Runtime/Loader/AssemblyLoadContext.cs

```diff
@@ -507,11 +507,11 @@ public ContextualReflectionScope EnterContextualReflection()
         /// Returns a disposable ContextualReflectionScope for use in a using block. When the using calls the
         /// Dispose() method, it restores the ContextualReflectionScope to its previous value.
         /// </remarks>
-        public static ContextualReflectionScope EnterContextualReflection(Assembly activating)
+        public static ContextualReflectionScope EnterContextualReflection(Assembly? activating)
         {
-            return activating != null ?
-                GetLoadContext(activating).EnterContextualReflection() :
-                new ContextualReflectionScope(null);
+            AssemblyLoadContext? assemblyLoadContext = (activating != null) ?
+                GetLoadContext(activating) : null;
+            return assemblyLoadContext?.EnterContextualReflection() ?? ContextualReflectionScope(null);
```

Comment:
Wouldn't it be better produced code to do

```cs
if (assemblyLoadContext == null)
     return assemblyLoadContext.EnterContextualReflection();

return ContextualReflectionScope(null);
```

Reply:
I am not a codegen expert.  It maybe more readable.